### DISCUSSION TITLE
problem with fresh install

### DISCUSSION THREAD

**Discussion by Rico29** [akvorado/akvorado#170] [Q&A] [answered]
Hello,
I've spent my day to try to install akvorado latest release with docker.
everything seems to work fine, except that the inlet container does not receive the flows. 

tcpdump on host receives the flows (udp/5022) and I can see that they are sent to docker akvorado-inlet container IP address

but curl -s http://127.0.0.1:8080/api/v0/inlet/flows\?limit=1 does not show anything.
nothing in web frontent
no snmp queries sent to the flows exporter.

iptables rules policy are in ACCEPT
Where should I look / how can I debug that ?

regards
Cédric

**Comment by vincentbernat** [comment]
Check the troubleshooting section of the documentation, notably to check the metrics to know when the flows are dropped. I suppose you replaced/added 5022 in `docker-compose.yml`, but you also need to do that in `akvorado.yaml`. Did you do that?

**Comment by Rico29** [comment]
Hello @vincentbernat
sorry for the late response.
I've kept 2055 everywhere, I miswrote 2055 to 5022. sorry for that;

I did not change the original port.

here's a quick debug output :

```
# tcpdump -i any 'udp dst port 2055' -nq
07:38:45.123479 br-09bb6f40b70a Out ifindex 13 02:42:5e:a6:26:1b 192.168.155.101.37672 > 172.20.0.3.2055: UDP, length 476
07:38:45.123480 veth92e9bb5 Out ifindex 280 02:42:5e:a6:26:1b 192.168.155.101.37672 > 172.20.0.3.2055: UDP, length 476
07:38:45.123486 eno1  In  ifindex 2 24:6e:96:54:03:34 192.168.155.101.37672 > 192.168.155.109.2055: UDP, length 136
07:38:45.123486 bond0 In  ifindex 8 24:6e:96:54:03:34 192.168.155.101.37672 > 192.168.155.109.2055: UDP, length 136
07:38:45.123490 br-09bb6f40b70a Out ifindex 13 02:42:5e:a6:26:1b 192.168.155.101.37672 > 172.20.0.3.2055: UDP, length 136
07:38:45.123491 veth92e9bb5 Out ifindex 280 02:42:5e:a6:26:1b 192.168.155.101.37672 > 172.20.0.3.2055: UDP, length 136
```

```

# docker ps | grep 2055
f23b96545d91   ghcr.io/akvorado/akvorado:latest    "/usr/local/bin/akvo…"   4 days ago   Up 4 days (healthy)   0.0.0.0:2055->2055/udp, 0.0.0.0:6343->6343/udp, 8080/tcp   akvorado-inlet

# docker inspect akvorado-inlet | grep -i ipaddr
            "SecondaryIPAddresses": null,
            "IPAddress": "",
                    "IPAddress": "172.20.0.3",

```
**troubleshooting :** 
```

# curl -s http://127.0.0.1:8080/api/v0/inlet/metrics | grep '^akvorado_inlet_flow_input_udp_packets'
akvorado_inlet_flow_input_udp_packets{exporter="192.168.155.101",listener="0.0.0.0:2055",worker="1"} 2.77424674e+08
akvorado_inlet_flow_input_udp_packets{exporter="192.168.155.101",listener="0.0.0.0:2055",worker="3"} 3.907401e+06


# curl -s http://127.0.0.1:8080/api/v0/inlet/metrics | grep '^akvorado_inlet' | grep _errors
akvorado_inlet_core_flows_errors{error="input and output interfaces missing",exporter="192.168.155.101"} 2.014346901e+09
akvorado_inlet_core_flows_errors{error="sampling rate missing",exporter="192.168.155.101"} 2.014346901e+09
akvorado_inlet_flow_decoder_netflow_errors_count{error="template not found",exporter="192.168.155.101"} 5
```


so it seems errors a related to a misconfigured exporter. I'm using pmacct. do you have a sample pmacct config ? I'll try to find a working config.

Regards
Cédric

**Reply by vincentbernat** [reply]
For the input/output interface, check with Wireshark what kind of packets you get. However, half of the packets seem to be have at least one of the interfaces because the sampling rate is checked next. For the second error, you need to configure a sampling rate (you can set it to 1) in pmacct (with `pmacctd_sampling_rate`). Alternatively, you can override the value in Akvorado configuration too.

**Reply by Rico29** [reply]
inputInt and outputInt are set to 0 in ipfix packets...
fixed the sampling_rate thing

**Reply by Rico29** [reply]
should it work correctly if one of the two interfaces = 0 ?
for example if I have : 
InputInt: 0
OutputInt : 51

Regards

**Reply by Rico29** [reply]
seems better with updated pmacct config file (see below) :

```
# curl -s http://127.0.0.1:8080/api/v0/inlet/metrics | grep '^akvorado_inlet' | grep _errors
akvorado_inlet_core_flows_errors{error="SNMP cache miss",exporter="192.168.155.101"} 1.756617e+06
akvorado_inlet_core_flows_errors{error="input and output interfaces missing",exporter="192.168.155.101"} 2.034938122e+09
akvorado_inlet_core_flows_errors{error="sampling rate missing",exporter="192.168.155.101"} 2.017229771e+09
akvorado_inlet_flow_decoder_netflow_errors_count{error="error decoding",exporter="192.168.155.101"} 3622
akvorado_inlet_flow_decoder_netflow_errors_count{error="template not found",exporter="192.168.155.101"} 5
```


it seems that akvorado's snmp queries :     192.168.155.109.52063 > 192.168.155.101.161:  { SNMPv2c C="xxxxx" { GetRequest(128) R=2075222723  .1.3.6.1.2.1.1.5.0 .1.3.6.1.2.1.2.2.1.2.52 .1.3.6.1.2.1.31.1.1.1.18.52 .1.3.6.1.2.1.31.1.1.1.15.52 .1
.3.6.1.2.1.2.2.1.2.51 .1.3.6.1.2.1.31.1.1.1.18.51 .1.3.6.1.2.1.31.1.1.1.15.51 } }

with 51 and 52 beeing the interface id I populate in ipfix packets...

**Reply by Rico29** [reply]
changed ifindexes in pretag map to the good ones, see pretag_map below.

**Comment by Rico29** [comment]
and here's my (updated) pmacctd exporter config file :
```
interface: bond0
aggregate: etype, tag, src_host, dst_host, src_port, dst_port, proto, tos, vlan

nfprobe_version: 10
plugins: nfprobe[ip]

nfprobe_receiver[ip]: 192.168.155.109:2055
nfprobe_timeouts[ip]: tcp=120:maxlife=3600

pmacctd_flow_lifetime: 30
sampling_rate: 10

pre_tag_map: /etc/pmacct/pretag.map
nfprobe_direction[ip]: tag
nfprobe_ifindex[ip]: tag2
```

pretag.map : 

```

set_tag=1 filter='ether dst 3c:fd:fe:a7:7f:68 && ether[14:2] & 0xfff == 50' jeq=cinqzero
set_tag=2 filter='ether src 3c:fd:fe:a7:7f:68 && ether[14:2] & 0xfff == 50' jeq=cinqzero

set_tag=1 filter='ether dst 3c:fd:fe:a7:7f:68 && ether[14:2] & 0xfff == 51' jeq=cinqun
set_tag=2 filter='ether src 3c:fd:fe:a7:7f:68 && ether[14:2] & 0xfff == 51' jeq=cinqun

set_tag=1 filter='ether dst 3c:fd:fe:a7:7f:68 && ether[14:2] & 0xfff == 52' jeq=cinqdeux
set_tag=2 filter='ether src 3c:fd:fe:a7:7f:68 && ether[14:2] & 0xfff == 52' jeq=cinqdeux


#iso.3.6.1.2.1.2.2.1.2.12 = STRING: "bond0.50"
#iso.3.6.1.2.1.2.2.1.2.13 = STRING: "bond0.51"
#iso.3.6.1.2.1.2.2.1.2.14 = STRING: "bond0.52"
#iso.3.6.1.2.1.2.2.1.2.15 = STRING: "bond1.40"


set_tag2=12 label=cinqzero
set_tag2=13 label=cinqun
set_tag2=14 label=cinqdeux

set tag2=999 filter='net 0.0.0.0/0'
```

**Comment by Rico29** [comment, marked as answer]
Everything seemms t work as expected. I'll try to find time to document the setup using pmacct as flow sender
thanks !

**Reply by Rico29** [reply]
@vincentbernat can I do a PR to update https://demo.akvorado.net/docs/operations#router-configuration with infos for pmacct ? if yes, where ?

**Reply by vincentbernat** [reply]
Documentation is here: https://github.com/akvorado/akvorado/tree/main/console/data/docs
